VMware.CloudFoundation.Reporting.psm1

```diff
@@ -824,7 +824,158 @@ Function Request-SoSHealthJson {
+    $scopePayload | Add-Member -notepropertyname 'includeAllDomains' -notepropertyvalue $false
+    $scopePayload | Add-Member -notepropertyname 'includeFreeHosts' -notepropertyvalue $false
+    $scopePayload.domains += $domainsArray
+    $healthSummarySpec = New-Object -TypeName psobject
+    $healthSummarySpec | Add-Member -notepropertyname 'healthChecks' -notepropertyvalue $healthChecksPayload
+    $healthSummarySpec | Add-Member -notepropertyname 'options' -notepropertyvalue $optionsPayload
+    $healthSummarySpec | Add-Member -notepropertyname 'scope' -notepropertyvalue $scopePayload
+    
+    # Request VCF Token
+    Request-VCFToken -fqdn $server -Username $user -Password $pass -skipCertificateCheck -ErrorAction SilentlyContinue -ErrorVariable ErrMsg | Out-Null
+
+    # Create VCF REST API Header
+    $vcfHeaders = @{"Content-Type" = "application/json" }
+    $vcfHeaders.Add("Authorization", "Bearer $accessToken")
+
+    Try {
+        if ($PsBoundParameters.ContainsKey("allDomains")) {
+            $healthSummarySpec.scope.includeAllDomains = $true
+            if ($PSEdition -eq "Core" -and ($PSVersionTable.OS).Split(' ')[0] -eq "Linux") {
+                $reportDestination = ($reportDestination = ($reportPath + "\" + $server.Split(".")[0] + "-all-health-results.json")).split('\') -join '/' | Split-Path -NoQualifier
+            } else {
+                $reportDestination = ($reportPath + "\" + $server.Split(".")[0] + "-all-health-results.json")
+            }
+        } elseif ($PsBoundParameters.ContainsKey("workloadDomain")) {
+            $healthSummarySpec.scope.domains[0].domainName = $workloadDomain
+            if ($PSEdition -eq "Core" -and ($PSVersionTable.OS).Split(' ')[0] -eq "Linux") {
+                $reportDestination = ($reportDestination = ($reportPath + "\" + $workloadDomain + "-all-health-results.json")).split('\') -join '/' | Split-Path -NoQualifier
+            } else {
+                $reportDestination = ($reportPath + "\" + $workloadDomain + "-all-health-results.json")
+            }
+        }
+        if (Test-VCFConnection -server $server) {
+            # Use Rest API method to request for health summary
+            $uri = "https://$server/v1/system/health-summary"
+            $healthSummaryPayload = ConvertTo-JSON $healthSummarySpec -depth 10
+            $response = Invoke-RestMethod -Method POST -URI $uri -headers $vcfHeaders -body $healthSummaryPayload
+			if ($response.id -eq "") {
+				write-error "Health Summary request run into issues please rerun the script"
+				return $false
+			}
+			$requestID = $response.id
+			
+            # Pulling request status
+            $uri = "https://$server/v1/system/health-summary/$requestID"
+            $response = Invoke-RestMethod -Method GET -URI $uri -headers $vcfHeaders
+            $escapeCounter = 0
+			while (($escapeCounter -lt 30) -and !($response.status -match "COMPLETED")) {
```

Comment:
```suggestion
			While (($escapeCounter -lt 30) -and !($response.status -match "COMPLETED")) {
```

Reply:
addressed in commit d64ce9ff50fe306dc3219c3013b0849ffc2e49c5